commit dcd9228b2911d5026dc50cc38b25829ee6664116
Author: gwx <wx.gu@netxeon.com>
Date:   Tue Mar 5 18:36:59 2019 +0800

    AV interface noise reduction lower
    
    	/system/media
           /common/
    
    Conflicts:
    	sound/soc/amlogic/auge/tdm.c
    
    Conflicts:
    	sound/soc/amlogic/auge/card.c
    	sound/soc/amlogic/auge/tdm.c

diff --git a/arch/arm/boot/dts/amlogic/g12b_a311d_w400.dts b/arch/arm/boot/dts/amlogic/g12b_a311d_w400.dts
index dd9e717..e2f1ff0 100644
--- a/arch/arm/boot/dts/amlogic/g12b_a311d_w400.dts
+++ b/arch/arm/boot/dts/amlogic/g12b_a311d_w400.dts
@@ -531,7 +531,7 @@
 
 		//aml-audio-card,aux-devs = <&amlogic_codec>;
 		/*avout mute gpio*/
-		avout_mute-gpios = <&gpio_ao GPIOAO_2 GPIO_ACTIVE_HIGH>;
+		avout_mute-gpios = <&gpio_ao GPIOAO_2 GPIO_ACTIVE_LOW>;
 		/*for audio effect ,eqdrc */
 		aml-audio-card,effect = <&audio_effect>;
 
diff --git a/sound/soc/amlogic/auge/card.c b/sound/soc/amlogic/auge/card.c
index f0b32d3..c8d1932 100644
--- a/sound/soc/amlogic/auge/card.c
+++ b/sound/soc/amlogic/auge/card.c
@@ -87,7 +87,7 @@ struct aml_card_data {
 
 	struct aml_chipset_info *chipinfo;
 };
-
+struct aml_card_data *priv;
 #define aml_priv_to_dev(priv) ((priv)->snd_card.dev)
 #define aml_priv_to_link(priv, i) ((priv)->snd_card.dai_link + (i))
 #define aml_priv_to_props(priv, i) ((priv)->dai_props + (i))
@@ -762,7 +762,12 @@ static int aml_card_parse_gpios(struct device_node *node,
 
 	return 0;
 }
+void gpiod_direction_outputao(int value){
 
+gpiod_direction_output(priv->avout_mute_desc,
+                              value);
+}
+EXPORT_SYMBOL_GPL(gpiod_direction_outputao);
 static void aml_init_work(struct work_struct *init_work)
 {
 	struct aml_card_data *priv = NULL;
@@ -873,7 +878,7 @@ static int aml_card_parse_of(struct device_node *node,
 
 static int aml_card_probe(struct platform_device *pdev)
 {
-	struct aml_card_data *priv;
+	/*struct aml_card_data *priv;*/
 	struct snd_soc_dai_link *dai_link;
 	struct aml_dai_props *dai_props;
 	struct device_node *np = pdev->dev.of_node;
diff --git a/sound/soc/amlogic/auge/tdm.c b/sound/soc/amlogic/auge/tdm.c
index ea098f2..d4dbb26 100644
--- a/sound/soc/amlogic/auge/tdm.c
+++ b/sound/soc/amlogic/auge/tdm.c
@@ -613,7 +613,7 @@ static int aml_dai_tdm_prepare(struct snd_pcm_substream *substream,
 
 	return 0;
 }
-
+extern void gpiod_direction_outputao(int value);
 static int aml_dai_tdm_trigger(struct snd_pcm_substream *substream, int cmd,
 			       struct snd_soc_dai *cpu_dai)
 {
@@ -672,6 +672,7 @@ static int aml_dai_tdm_trigger(struct snd_pcm_substream *substream, int cmd,
 				&& p_tdm->en_share) {
 				aml_spdifout_mute_without_actrl(0, false);
 			}
+			gpiod_direction_outputao(0);
 		} else {
 			dev_info(substream->pcm->card->dev, "tdm capture enable\n");
 			aml_toddr_enable(p_tdm->tddr, 1);
@@ -725,6 +726,7 @@ static int aml_dai_tdm_trigger(struct snd_pcm_substream *substream, int cmd,
 				aml_frddr_check(p_tdm->fddr);
 
 			aml_frddr_enable(p_tdm->fddr, false);
+			gpiod_direction_outputao(1);
 		} else {
 			bool toddr_stopped = false;
 
